riscv/tests/riscv_data/function_pointer/src/lib.rs

```diff
@@ -0,0 +1,30 @@
+#![no_std]
+
+use powdr_riscv_runtime::get_prover_input;
+
+// Never inline to make sure the function is not optimized away, and public to
+// make sure op argument is not removed, forcing a function pointer to be loaded
+// and passed.
+#[inline(never)]
+pub fn apply_op(op: fn(u32, u32) -> u32, a: u32, b: u32) -> u32 {
```

Comment:
Wouldn't it be better to force dynamic dispatch?

Reply:
Anyway, this commit was already merged in another PR. I made this branch on top of another branch and forgot to set the target appropriately. Sorry.